scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Connection.java

```diff
@@ -491,13 +511,15 @@ public void sessionEstablished(Handshaker handshaker, DTLSSession session) throw
 		@Override
 		public void handshakeCompleted(Handshaker handshaker) {
 			if (ongoingHandshake.compareAndSet(handshaker, null)) {
+				startedByClient.set(null);
```

Comment:
Reset the random has the same effect as ending the handshake in the implementation before. 
With 32 bytes random, I'm not sure, if this should be done at all or at least delayed. 
With a CLIENT_HELLO duplicate "on the way" and a reset random, a new handshake may be trigger instead of ignoring the duplicate.